node/src/types/deploy.rs

```diff
@@ -1123,6 +1130,15 @@ mod tests {
         check_is_not_valid(deploy, DeployValidationFailure::InvalidDeployHash);
     }
 
+    #[test]
```

Comment:
@rafal-ch Is this what you meant by test for a deploy with no approvals

Reply:
Not exactly, sorry for not being clear enough :)

This is an unit tests that tests the validation on the type level. Meaning, it tests whether the `Deploy` can correctly validate itself. This is all good.

It, however, doesn't validate if the `is_valid()` function is at all used when the real deploy is being created. So the test I proposed to consider would have steps like:

1. setup and run test network
2. create deploy with no approvals
3. try to deploy the deploy :-)
4. observe an error

**But again, this could be too much of a hassle if we don't have a framework for such tests ready.**

PS. 
What I see the call chain to accept the deploy is the following:
* `DeployAcceptor::handle_event()`
* `DeployAcceptor::accept()`
* `Deploy::is_acceptable()`
* `Deploy::is_valid()`
and since I cannot locate any tests for `DeployAcceptor` I feel a bit anxious :)